src/nvim/os/shell.c

```diff
@@ -37,6 +37,40 @@ typedef struct {
 #endif
 
 
+/// Process command string with 'shellxescape' (p_sxe) and 'shellxquote'
+/// (p_sxq)
+///
+/// @param cmd Command string
+/// @return A newly allocated command string. It must be freed with
+///         `xfree` when no longer needed.
+static char *cmd_sxe_sxq(const char *cmd)
+{
+  char *ncmd;
+
+  if (cmd == NULL || *p_sxq == NUL) {
+    ncmd = (cmd == NULL) ? NULL : xstrdup(cmd);
+  } else {
+    char *ecmd = (char *)cmd;
+
+    if (*p_sxe != NUL && STRCMP(p_sxq, "(") == 0) {
+      ecmd = (char *)vim_strsave_escaped_ext((char_u *)cmd, p_sxe, '^', false);
+    }
+    ncmd = xmalloc(strlen(ecmd) + STRLEN(p_sxq) * 2 + 1);
+    STRCPY(ncmd, p_sxq);
+    strcat(ncmd, ecmd);
+    /* When 'shellxquote' is ( append ).
+     * When 'shellxquote' is "( append )". */
```

Comment:
Maybe add commas and quotes (i.e. `When 'shellquote' is '(', append ')'.`) to increase readability. Also, please use `//` comment style.